lib/miq_environment.rb

```diff
@@ -24,6 +24,10 @@ def self.supports_nohup_and_backgrounding?
       @supports_nohup = is_appliance? && supports_command?('nohup')
     end
 
+    def self.is_production_build?
```

Comment:
We can debate the name...I was having a hard time with this.

Note that in discussions with @simaishi, it might be preferable to have a dedicated ENV var for this, such as `MANAGEIQ_IMAGE=true`, which would be present in all build artifacts whether appliance, podified, or container.  Then, this becomes a simple check for that ENV VAR, and all of the other checks in this file can use that value as a an extra sanity check.

Reply:
If we honored `RAILS_SERVE_STATIC_FILES`'s value, these types of installations can set this env variable and this method is not needed.   Thoughts?

Reply:
RAILS_SERVE_STATIC_FILES is meant for _dev_ use case of testing locally when Rails.env == "production"...that is, devs are expected to set it.   The rationale is that in production Rails static asset serving is off by default, and then you opt-in with the env var, so on real production you specifically don't set anything.